qiskit/quantum_info/operators/scalar_op.py

```diff
@@ -0,0 +1,264 @@
+"""
+
+from numbers import Number
+import numpy as np
+
+from qiskit.exceptions import QiskitError
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+from qiskit.quantum_info.operators.operator import Operator
+
+
+class ScalarOp(BaseOperator):
+    """Scalar identity operator class.
+
+    This is a symbolic representation of an scalar identity operator on
+    multiple subsystems. It may be used to initialize a symbolic scalar
+    multiplication of an identity and then be implicitly converted to other
+    kinds of operator subclasses by using the :meth:`compose`, :meth:`dot`,
+    :meth:`tensor`, :meth:`expand` methods.
+    """
+
+    def __init__(self, dims, coeff=None):
+        """Initialize an operator object.
+
+        Args:
+            dims (int or tuple): subsystem dimensions.
+            coeff (complex or None): an optional scalar coefficient for
+                                     the identity operator (Default: None).
+
+        Raises:
+            QiskitError: If the optional coefficient is invalid.
+        """
+        input_dims = self._automatic_dims(dims, np.product(dims))
+        if coeff is not None and not isinstance(coeff, Number):
+            raise QiskitError("Coeff {} must be None or a number.".format(coeff))
+        self._coeff = coeff
+        super().__init__(input_dims, input_dims)
+
+    def __repr__(self):
+        if self.coeff is None:
+            return 'ScalarOp({})'.format(self._input_dims)
+        return 'ScalarOp({}, coeff={})'.format(
+            self._input_dims, self.coeff)
+
+    @property
+    def coeff(self):
+        """Return the coefficient"""
+        return self._coeff
+
+    def conjugate(self):
+        """Return the conjugate of the operator."""
+        if self.coeff is None:
+            return self
+        ret = self.copy()
+        ret._coeff = np.conjugate(self.coeff)
+        return ret
+
+    def transpose(self):
+        """Return the transpose of the operator."""
+        return self
+
+    def is_unitary(self, atol=None, rtol=None):
+        """Return True if operator is a unitary matrix."""
+        if self.coeff is None:
+            return True
+        if atol is None:
+            atol = self._atol
+        if rtol is None:
+            rtol = self._rtol
+        return np.isclose(np.abs(self.coeff), 1, atol=atol, rtol=rtol)
+
+    def to_matrix(self):
+        """Convert to a Numpy matrix."""
+        dim, _ = self.dim
+        iden = np.eye(dim, dtype=complex)
+        if self.coeff is None:
+            return iden
+        return self.coeff * iden
+
+    def to_operator(self):
+        """Convert to an Operator object."""
+        return Operator(self.to_matrix(),
+                        input_dims=self._input_dims,
+                        output_dims=self._output_dims)
+
+    def tensor(self, other):
+        """Return the tensor product operator self ⊗ other.
+
+        Args:
+            other (BaseOperator): an operator object.
```

Comment:
can you tensor a ScalarOp with a QuantumChannel? If not this should just be Operator I think (would help if ScalarOp was derived from Operator like I commented earlier)

Reply:
Yes you can tensor with QuantumChannel. the idea is you can tensor with anything derived from BaseOperator